src/main/java/io/opentracing/contrib/global/GlobalTracer.java

```diff
@@ -0,0 +1,184 @@
+package io.opentracing.contrib.global;
+
+import io.opentracing.NoopTracerFactory;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.global.concurrent.TracedCallable;
+import io.opentracing.contrib.global.concurrent.TracedRunnable;
+import io.opentracing.propagation.Format;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The {@linkplain GlobalTracer} forwards all methods to a single {@link Tracer} implementation that can be
+ * instantiated in one of two ways:
+ * <ol>
+ * <li>Explicitly, by calling {@link #setTracer(Tracer)} with a configured tracer implementation, or:</li>
+ * <li>Automatically by using the Java <code>ServiceLoader</code> SPI mechanism to load an implementation
+ * from the classpath.</li>
+ * </ol>
+ *
+ * @author Sjoerd Talsma
+ */
+public final class GlobalTracer implements Tracer {
+    private static final Logger LOGGER = Logger.getLogger(GlobalTracer.class.getName());
+
+    private static final GlobalTracer INSTANCE = new GlobalTracer();
+
+    /**
+     * The resolved {@link Tracer} to delegate the global tracing implementation to.<br>
+     * This can be either an {@link #setTracer(Tracer) explicitly set delegate}
+     * or the automatically resolved Tracer implementation.<br>
+     * Management of this reference is the responsibility of the {@link #lazyTracer()} method.
+     */
+    private final AtomicReference<Tracer> globalTracer = new AtomicReference<Tracer>();
```

Comment:
I don't want to make things accessible just for testing purposes.
I agree however that the loadSingleton method has too many concerns.
I'll create a separate issue for it and also add concurrency tests.